initrd/options.go

```diff
@@ -4,29 +4,69 @@
 // You may not use this file except in compliance with the License.
 package initrd
 
+import "fmt"
+
 type InitrdOptions struct {
-	compress bool
-	output   string
-	cacheDir string
-	arch     string
-	workdir  string
+	arch       string
+	cacheDir   string
+	compress   bool
+	keepOwners bool
+	output     string
+	fsType     FsType
+	workdir    string
 }
 
 type InitrdOption func(*InitrdOptions) error
 
-// WithCompression sets the compression of the resulting CPIO archive file.
-func WithCompression(compress bool) InitrdOption {
+type FsType string
+
+const (
+	FsTypeCpio    = FsType("cpio")
+	FsTypeErofs   = FsType("erofs")
+	FsTypeUnknown = FsType("unknown")
+)
+
+var _ fmt.Stringer = (*FsType)(nil)
+
+// String implements fmt.Stringer
+func (fsType FsType) String() string {
+	return string(fsType)
+}
+
+// FsTypes returns the list of possible fsTypes.
+func FsTypes() []FsType {
+	return []FsType{
+		FsTypeCpio,
+		FsTypeErofs,
+	}
+}
+
+// FsTypeNames returns the string representation of all possible
+// fsType implementations.
+func FsTypeNames() []string {
+	types := []string{}
+	for _, name := range FsTypes() {
+		types = append(types, name.String())
+	}
+
+	return types
+}
+
+// WithArchitecture sets the architecture of the file contents of binaries in
+// the initramfs.  Files may not always be architecture specific, this option
+// simply indicates the target architecture if any binaries are compiled by the
+// implementing initrd builder.
+func WithArchitecture(arch string) InitrdOption {
 	return func(opts *InitrdOptions) error {
-		opts.compress = compress
+		opts.arch = arch
 		return nil
 	}
 }
 
-// WithOutput sets the location of the output location of the resulting CPIO
-// archive file.
-func WithOutput(output string) InitrdOption {
+// WithCompression sets the compression of the resulting CPIO archive file.
+func WithCompression(compress bool) InitrdOption {
 	return func(opts *InitrdOptions) error {
-		opts.output = output
+		opts.compress = compress
```

Comment:
It looks like you re-ordered the file, this makes git blame sad, can you please keep the function order?

Reply:
I ordered* as it was not ordered before

I can just restore the old order